File: engine/src/main/java/de/hanno/hpengine/engine/model/loader/md5/AnimationController.java
package de.hanno.hpengine.engine.model.loader.md5;

public class AnimationController {
    final AnimationState animationState = new AnimationState();

    public AnimationController(int maxFrames) {
        this.animationState.maxFrames = maxFrames;
    }

    public void nextFrame() {
        animationState.nextFrame();
    }

    public boolean isHasUpdated() {
        return animationState.isHasUpdated();
    }
    public void setHasUpdated(boolean hasUpdated) {
        animationState.setHasUpdated(hasUpdated);
    }

    public int getCurrentFrameIndex() {
        return animationState.currentFrame;
    }

}

File: engine/src/main/java/de/hanno/hpengine/engine/model/loader/md5/AnimationController.kt
package de.hanno.hpengine.engine.model.loader.md5

data class AnimationController(val maxFrames: Int, val fps: Float) {
    val animationState : AnimationState = AnimationState(maxFrames, fps)

    fun update(seconds: Float) {
        animationState.update(seconds)
    }

    var isHasUpdated: Boolean
        get() = animationState.isHasUpdated
        set(hasUpdated) {
            animationState.isHasUpdated = hasUpdated
        }

    val currentFrameIndex: Int
        get() = animationState.currentFrame

}